alot/commands/__init__.py

```diff
@@ -15,25 +15,44 @@ class Command(object):
 
     """base class for commands"""
     repeatable = False
+    mode = ""
+    cmdname = ""
 
     def __init__(self):
         self.undoable = False
         self.help = self.__doc__
 
-        class_ = type(self)
-        class_name = class_.__name__
-        cmdname, mode = reverse_lookup_command(class_)
-
-        # fetch and set pre and post command hooks
-        # they are set to None if not defined in the hooks file
-        get_hook = settings.get_hook
-        self.prehook = (get_hook('pre_%s_%s' % (mode, cmdname)) or
-                        get_hook('pre_global_%s' % cmdname))
-        self.posthook = (get_hook('post_%s_%s' % (mode, cmdname)) or
-                         get_hook('post_global_%s' % cmdname))
+        # set to False if no attemtp to fetch hooks from settings
+        self._prehook = False 
+        self._posthook = False
 
-        logging.debug("%s prehook: %s", class_name, str(self.prehook))
-        logging.debug("%s posthook: %s", class_name, str(self.posthook))
+    @property
+    def id(self):
+        """Mode and command name of instance as string."""
+        class_ = type(self)
+        return class_.mode + '.' + class_.cmdname
```

Comment:
Gah, python3 can have `property` and `classmethod` used together, which would be nice. You could implement a `classproperty` in the helpers module that would make this nicer. Something like:

```python
class classproperty(object):
    """A decorator like `property` but works on classes."""
    def __init__(self, func):
        self.__func = func

    def __get__(self, instance, cls):
        return self.__func(cls)
```

Then you could implement this method as:
```python
@classproperty
def id(cls):
    return '{}.{}'.format(cls.mode, cls.cmdname)
```
However, this could be done as a followup instead of in this patch.